pipeline_dp/combiners.py

```diff
@@ -213,13 +229,29 @@ def merge_accumulators(
         return (privacy_id_count1 + privacy_id_count2,
                 tuple(merged_accumulators))
 
-    def compute_metrics(
-            self, compound_accumulator: AccumulatorType) -> Tuple[int, list]:
+    def compute_metrics(self,
+                        compound_accumulator: AccumulatorType) -> MetricsTuple:
         privacy_id_count, accumulator = compound_accumulator
-        metrics = []
+        privacy_id_count, count, sum, mean = None, None, None, None
         for combiner, acc in zip(self._combiners, accumulator):
-            metrics.append(combiner.compute_metrics(acc))
-        return (privacy_id_count, metrics)
+            if isinstance(combiner, PrivacyIdCountCombiner):
```

Comment:
Checking types violates polymorphism, ideally it would be able not to check types of combiners. 

For example we can do in the following way:
1.Introduce method

```
class Combiner

@abstract
def metrics_names():
   pass
```

2.Each specific combiner knows which metrics needs to be returned, e.g. 
```
class SumCombiner:
    def metrics_names(): 
         return ["SUM"]

class MeanCombiner
      def  metrics_names(): 
          # depends on AggregateParams.Metrics
```

3.`MeanCombiner.compute_metrics returns` only metrics requested in` AggregateParams.Metrics`  (i.e. we're shifting resolving which metrics to return to MeanCombiner)

4.CompoundCombiner  simply concatenates metrics from the internal combinders

WDYT?